### PR TITLE
[DEV APPROVED] - Run the tests in Jenkins under Docker

### PR DESCRIPTION
**Pull request by norm**
In order to run the tests for this gem in Jenkins, we need to provide
a Docker image. It runs Ubuntu 17.04, so the version of ruby is upgraded
to match the default in that OS. In addition, the version of rake is
upgraded so that `bundle install` succeeds.

### CODE REVIEW

**Comment by norm** [comment]
(as you can see if you open the "All checks have passed" block by clicking "Show all checks", Jenkins has successfully run the tests already)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Docker image to run the tests in Jenkins.
+
+FROM ubuntu:17.04
+MAINTAINER development.team@moneyadviceservice.org.uk
+
+# Install needed packages.
+RUN apt-get update -q \
+    && DEBIAN_FRONTEND="noninteractive" apt-get upgrade \
+            -qy \
+            -o Dpkg::Options::="--force-confnew" \
```
Comment (tomas-stefano) [inline Dockerfile]: What is "--force-confnew" means?

**Comment by tomas-stefano** [comment]
Just a small question but LGTM 👍 👍

File: Dockerfile (hunk 1 above)
Comment (norm) [inline Dockerfile]: It means that if a package is upgraded that contains example configuration files as part of the package, install that configuration file if it has changed. Basically "give me the latest software and its default config files".

**Comment by pmcostello** [comment]
👍

**Pull request merged**